### ISSUE TITLE
MySqlPlatform::getListTableColumnsSQL() is not escaping names

### ISSUE THREAD

**Issue by sadortun** [doctrine/dbal#2436]
Hi !

When creating by accident a table that contain a `\` in the name, `abc\` The migration fail to read and escape correctly the table names.

$ php app/console oro:platform:update --env dev --force
Process migrations...

```
[Doctrine\DBAL\Exception\SyntaxErrorException]
An exception occurred while executing 'SELECT COLUMN_NAME AS Field, COLUMN_TYPE AS Type, IS_NULLABLE AS `Null`, COLUMN_KEY AS `Key`, COLUMN_DEFAULT AS `Default`, EXTRA
 AS E  xtra, COLUMN_COMMENT AS Comment, CHARACTER_SET_NAME AS CharacterSet,  OLLATION_NAME AS Collation FROM information_schema.COLUMNS WHERE TABLE_SCHEMA = 'asdf' AND TABLE   _NAME = ''abc\'':

SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the ri
ght syntax to use near ''oro_tag_taxonomy\'' at line 1
```

**Comment by Deltachaos** [comment]
@sadortun i think this has todo with the behaivor of the quote strategy which is not escaping reserved fieldnames per default. This behaivor may change due to doctrine/doctrine2#5903.
Beside that i think migrations are a feature of the ORM.

Or i am wrong @Majkl578 @deeky666

**Comment by sadortun** [comment]
@Deltachaos Thanks. Please me know if you believe it's an internal issue, or an issue with the library client. If it's the case i'll re-open the bug on their side

**Comment by deeky666** [comment]
This is an edge that has to be dealt with in DBAL. This is not about reserved keywords but about escape characters in column names passed into the metadata SQL (it seems).

**Comment by deeky666** [comment]
Fix supplied in #2442
